src/sql/engine/expr/ob_expr_time.cpp

```diff
@@ -218,10 +220,18 @@ int ObExprTimeBase::calc(const ObExpr &expr, ObEvalCtx &ctx, ObDatum &expr_datum
       } else {
         int idx = ot.parts_[type] - 1;
         if(0 <= idx  && idx < 7) {
-          const ObString &name = daynames[idx];
-          if (OB_FAIL(ObExprUtil::set_expr_ascii_result(expr, ctx, expr_datum, name))) {
-            LOG_WARN("failed to exec set_expr_ascii_result", K(expr), K(ctx), K(expr_datum), K(name));
-          }          
+           // const ObString &name = daynames[idx];
+          if(OB_FAIL(session->get_locale_name(locale_name))){
+              LOG_WARN("failed to get locale time name", K(expr), K(expr_datum));
+          } else {
+            OB_LOCALE *ob_cur_locale = ob_locale_by_name(locale_name);
+            OB_LOCALE_TYPE *locale_type = ob_cur_locale->day_names;
+            const char ** locale_daynames = locale_type->type_names_;
```

Comment:
in here do not split mysql mode with oracle, does u sure it is right in oceanbase oracle mode?

Reply:
> in here do not split mysql mode with oracle, does u sure it is right in oceanbase oracle mode?

Are you talking about method “ob_locale_by_name”？